docs/source/dataframe-design.rst

```diff
@@ -0,0 +1,109 @@
+Internal Design
+===============
+
+Dask dataframes coordinate many Pandas arrays arranged along an index. We
+define a ``dask.dataframe`` object with the following components:
+
+- A dask graph with a special set of keys designating partitions, such as
+  ``('x', 0), ('x', 1), ...``.
```

Comment:
I just copied that from the equivalent array docs, happy to go either way. This is a complete specification though, which is nice in its own way.